video/beta/src/main/java/com/example/video/TextDetection.java

```diff
@@ -0,0 +1,168 @@
+import com.google.cloud.videointelligence.v1p2beta1.NormalizedVertex;
+import com.google.cloud.videointelligence.v1p2beta1.TextAnnotation;
+import com.google.cloud.videointelligence.v1p2beta1.TextFrame;
+import com.google.cloud.videointelligence.v1p2beta1.TextSegment;
+import com.google.cloud.videointelligence.v1p2beta1.VideoAnnotationResults;
+import com.google.cloud.videointelligence.v1p2beta1.VideoIntelligenceServiceClient;
+import com.google.protobuf.ByteString;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+public class TextDetection {
+
+  // [START video_detect_text_beta]
+  /**
+   * Detect text in a video.
+   *
+   * @param filePath the path to the video file to analyze.
+   */
+  public static VideoAnnotationResults detectText(String filePath) throws Exception {
+
+    try (VideoIntelligenceServiceClient client = VideoIntelligenceServiceClient.create()) {
+
+      // Read file
+      Path path = Paths.get(filePath);
+      byte[] data = Files.readAllBytes(path);
+
+      // Create the request
+      AnnotateVideoRequest request = AnnotateVideoRequest.newBuilder()
+          .setInputContent(ByteString.copyFrom(data))
+          .addFeatures(Feature.TEXT_DETECTION)
+          .build();
+
+      // asynchronously perform object tracking on videos
+      OperationFuture<AnnotateVideoResponse, AnnotateVideoProgress> response =
+          client.annotateVideoAsync(request);
+
+      System.out.println("Waiting for operation to complete...");
+      // The first result is retrieved because a single video was processed.
+      VideoAnnotationResults result = response.get(300, TimeUnit.SECONDS)
+          .getAnnotationResults(0);
+
+      // Get only the first annotation for demo purposes.
+      TextAnnotation textAnnotation = result.getTextAnnotations(0);
+
+      System.out.println("Text: " + textAnnotation.getText());
+
+      // Get the first text segment.
+      TextSegment textSegment = textAnnotation.getSegments(0);
+
+      System.out.println(String.format("Start time: %.2f",
+          textSegment.getSegment().getStartTimeOffset().getSeconds()
+              + textSegment.getSegment().getStartTimeOffset().getNanos() / 1e9));
+
+      System.out.println(String.format("End time: %.2f",
+          textSegment.getSegment().getEndTimeOffset().getSeconds()
+              + textSegment.getSegment().getEndTimeOffset().getNanos() / 1e9));
+
+      System.out.println("Confidence: " + textSegment.getConfidence());
+
+      // Show the first result for the first frame in the segment.
+      TextFrame textFrame = textSegment.getFrames(0);
+
+      System.out.println("Time offset for the first frame: "
+          + (textFrame.getTimeOffset().getSeconds()
+          + textFrame.getTimeOffset().getNanos() / 1e9));
+
+      System.out.println("Rotated Bounding Box Vertices:");
+      for (NormalizedVertex normalizedVertex :
```

Comment:
This is already two lines, but easier to read: 
```suggestion
      List<NormalizedVertex> vertices = textFrame.getRotatedBoundingBox().getVerticesList()
      for (NormalizedVertex normalizedVertex : vertices) {
```

Reply:
Fixed